samza-core/src/main/scala/org/apache/samza/coordinator/JobModelManager.scala

```diff
@@ -365,6 +366,44 @@ class JobModelManager(
     }
   }
 
+  private def populateContainerLocalityMappings(): Unit = {
+    val allMappings = localityManager.readContainerLocality
+    import scala.collection.JavaConversions._
+    for (containerId <- jobModel.getContainers.keySet) {
+      if (allMappings.containsKey(containerId)) localityMappings.put(containerId, allMappings.get(containerId).get(SetContainerHostMapping.HOST_KEY))
+      else localityMappings.put(containerId, null)
+    }
+  }
+
+  def getAllContainerLocality: util.Map[String, String] = {
+    if (localityManager != null) populateContainerLocalityMappings
+    else if(localityMappings.isEmpty) jobModel.getContainers.keys.foreach(containerId => localityMappings.put(containerId, null))
+
+    localityMappings
+  }
+
+
+  def getContainerToHostValue(containerId: String, key: String): String = {
+    val EMPTY_STRING: String = ""
+    if (localityManager == null) return EMPTY_STRING
+    val mappings = localityManager.readContainerLocality.get(containerId)
+    if (mappings == null) return EMPTY_STRING
+    if (!mappings.containsKey(key)) return EMPTY_STRING
+    mappings.get(key)
+  }
+
+  def getAllContainerToHostValues(key: String): util.Map[String, String] = {
+    if (localityManager == null)
+      return new util.HashMap[String, String]()
+    val allValues = new util.HashMap[String, String]
+    import scala.collection.JavaConversions._
+    for (entry <- localityManager.readContainerLocality.entrySet) {
```

Comment:
Can we rename these to meaningful ones? It is hard to know what entry, key, value(key inside value) corresponds to?